### PR TITLE
#6 facebook auth

### PR DESCRIPTION
**Pull request by f1nwe**
- [ ] @Kmalynss 
- [ ] @denys281
- [x] @arsenlol 
This branch based on last commit in  #2_adding_authorization.
Added authentification via facebook to users model, "name" to Users model.
Also  corrected the errors observed in #2_adding_authorization pull request

### CODE REVIEW

File: app/views/home/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<h1>Home#index</h1>
+<p>Find me in app/views/home/index.html.erb</p>
+
+<% if current_user.nil? %>
+    <%= link_to new_user_session_path, class: 'login-button' do %>Sign in
+    <% end %>
```
Comment (denys281) [inline app/views/home/index.html.erb]: Fix formatting

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
-  </head>
+  <%= stylesheet_link_tag 'application', media: 'all' %>
+  <%= javascript_include_tag 'application' %>
+</head>
 
-  <body>
-    <%= yield %>
-  </body>
+<body>
+<%= yield %>
```
Comment (denys281) [inline app/views/layouts/application.html.erb]: Fix formatting

File: .idea/.generators [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
...eratorsGroup><Generator name="active_record:migration" /><Generator name="active_record:model" /><Generator name="active_record:observer" /><Generator name="active_record:session_migration" /><Generator name="controller" /><Generator name="erb:controller" /><Generator name="erb:mailer" /><Generator name="erb:scaffold" /><Generator name="generator" /><Generator name="helper" /><Generator name="integration_test" /><Generator name="mailer" /><Generator name="metal" /><Generator name="migration" /><Generator name="model" /><Generator name="model_subclass" /><Generator name="observer" /><Generator name="performance_test" /><Generator name="plugin" /><Generator name="resource" /><Generator name="scaffold" /><Generator name="scaffold_controller" /><Generator name="session_migration" /><Generator name="stylesheets" /><Generator name="test_unit:controller" /><Generator name="test_unit:helper" /><Generator name="test_unit:integration" /><Generator name="test_unit:mailer" /><Generator name="test_unit:model" /><Generator name="test_unit:observer" /><Generator name="test_unit:performance" /><Generator name="test_unit:plugin" /><Generator name="test_unit:scaffold" /></GeneratorsGroup></Settings>
```
Comment (denys281) [inline .idea/.generators]: Delete all .idea file

File: app/views/layouts/mailer.html.erb [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+  <style>
+    /* Email styles need to be inline */
+  </style>
+</head>
 
-  <body>
-    <%= yield %>
-  </body>
+<body>
+<%= yield %>
```
Comment (denys281) [inline app/views/layouts/mailer.html.erb]: Fix indentation

File: spec/controllers/omniauth_callbacks_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+describe Users::OmniauthCallbacksController, type: :controller do
+  before(:all) do
+    @successfull_auth_text = 'Successfully authenticated from Facebook account.'
+  end
+  describe 'when facebook email doesn\'t exist in the system' do
+    before(:each) do
+      env_for_omniauth
+      get :facebook
+      @user = User.find_by(email: 'vasya@pup.kin')
```
Comment (denys281) [inline spec/controllers/omniauth_callbacks_controller_spec.rb]: use let instead of instance variable

http://betterspecs.org/#let

Comment (igorkasyanchuk) [inline spec/controllers/omniauth_callbacks_controller_spec.rb]: but `let!(:user)` ... with explamation

File: app/views/layouts/mailer.html.erb (hunk 4 above)
Comment (igorkasyanchuk) [inline app/views/layouts/mailer.html.erb]: maybe let use haml/slim?

File: app/assets/stylesheets/debug_frame.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
+/* mixins, variables, etc. */
+
+$gray-medium-light: #eaeaea;
```
Comment (igorkasyanchuk) [inline app/assets/stylesheets/debug_frame.scss]: I would suggest to move all variables to variables.css.scss file and include in each file where you need

**Pull request closed without merging**